buildpacks/ruby/src/layers/ruby_install_layer.rs

```diff
@@ -130,10 +131,18 @@ fn download_url(stack: &StackId, version: impl std::fmt::Display) -> Result<Url,
     let base = "https://heroku-buildpack-ruby.s3.us-east-1.amazonaws.com";
     let mut url = Url::parse(base).map_err(RubyInstallError::UrlParseError)?;
 
-    url.path_segments_mut()
-        .map_err(|()| RubyInstallError::InvalidBaseUrl(String::from(base)))?
-        .push(stack)
-        .push(&filename);
+    if stack == &stack_id!("heroku-24") {
+        url.path_segments_mut()
+            .map_err(|()| RubyInstallError::InvalidBaseUrl(String::from(base)))?
+            .push(stack)
+            .push("amd64")
```

Comment:
This is a great starting point. The only thing I would change is to use the architecture of the current machine. We're doing this in classic via shelling out https://github.com/heroku/heroku-buildpack-ruby/blob/fd6585411cbf86d363ad0824f53d8edba1d7afed/lib/language_pack/base.rb#L45-L57.

I'm hoping on a plane in a few hours and then out tomorrow, if you're available to take a stab at adding that extra logic that would be amazing 🙏🏻. If not, I can take this commit and add that on top.

I was working towards upgrading libcnb first, but seeing how compact this PR is, I think I like the idea of merging this in now and cutting a builder so people can play with it.

Reply:
Yep shelling out is definitely not the best approach here.. This is somewhat related to https://github.com/heroku/buildpacks-nodejs/pull/814#discussion_r1580063265 -- e.g. whether to use the CNB targets, or the compile-time conditional (or `std::env::consts::ARCH` as in the node.js sample).

I alluded to a conversation I had with @Malax on this (re env provided target arch vs compile-time conditionals). What do you think about this @Malax (and did I remember the conversation correctly :-))?

Reply:
For sure, seems like it'd make a lot of sense to also use `context.target` in this buildpack when the buildpack is updated to newer `libcnb.rs` and `targets`.

I've updated this implementation to also factor in the current arch -- should be fairly straightforward to adapt this to use the `Target` instead.